fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/CascadesPlanner.java

```diff
@@ -363,10 +364,11 @@ private RecordQueryPlan resultOrFail() {
         final RelationalExpression singleRoot = currentRoot.getMembers().iterator().next();
         if (singleRoot instanceof RecordQueryPlan) {
             if (logger.isDebugEnabled()) {
-                logger.debug(KeyValueLogMessage.of("explain of plan",
+                logger.debug(KeyValueLogMessage.of("GML explain of plan",
                         "explain", PlannerGraphProperty.explain(singleRoot)));
+                logger.debug(KeyValueLogMessage.of("string explain of plan",
+                        "explain", ExplainPlanVisitor.toStringForDebugging((RecordQueryPlan)singleRoot)));
```

Comment:
Should this be consolidated into a single log message with both the GML and the string explain? Arguably, this should just have the string explain, as I think that will be more useful for humans reading through the logs, if we even want to rely on this log line at all.

Reply:
There is not enough testing for GML that's why it's there in the first place. Maybe I should open a ticket for that. For now I think this is safer and will not ever be triggered other than in local debugging sessions.

Reply:
https://github.com/FoundationDB/fdb-record-layer/issues/3017

Reply:
Okay, got it. But would it make sense to collapse both calls to a single log message, like:

```java
logger.debug(KeyValueLogMessage.of("explain of plan",
    "explain_gml", PlannerGraphProperty.explain(singleRoot),
    "explain_string", ExplainPlanVisitor.toStringForDebugging((RecordQueryPlan)singleRoot)));
```

I think you get the same amount of coverage, but only one log message. Which is something we sort of care about, even for debug logs